### ISSUE TITLE
Sia (SC) New Protocol Submission

### ISSUE THREAD

**Issue by hakkane84** [web3index/web3index-org#51]
## Protocol Name

Sia

## Symbol

SC

## Description

Sia is a leading decentralized cloud storage platform. No signups, no servers, no trusted third parties. Sia leverages blockchain technology to create a data storage marketplace that is more robust and more affordable than traditional cloud storage providers. Renters and hosts (the storage providers) reach storage agreements on the blockchain and transfer data without intermediaries in a truly decentralized fashion. The uploaded data is encrypted and distributed among multiple hosts redundantly, ensuring data durability and performance. More details about the technology can be found on https://sia.tech/technology.

The development of Sia started on 2014 and its blockchain was launched on 6 June 2015. Sia had a working storage network and marketplace since the genesis block. On 2020 the core development team released the layer-2 `Skynet` technology which has become a leading edge on web3 development. Skynet uses [public portals](https://siasky.net/) as gateways for interacting with the Sia hosts without the friction created by tokens or blockchains. Importantly, the data uploaded by one portal can be accessed with the same hash and address from any other public portal or through a private local Skynet client, ensuring unstoppable and censor-resistant applications. Skynet is used today not just for file transfer or as a CDN, but also for distributing and using dapps directly on the web browser. Dapps can even make use of a common decentralized identity system (called MySky).

Other third-party developments are leveraging the Sia storage network for offering storage services, as the enterprise-oriented and S3-compatible [Filebase](https://filebase.com/) or the file sharing service [Pixeldrain](https://pixeldrain.com/).

## Fee / revenue data source

Revenue data will be provided by a publicly accessible API endpoint provided by the third-party website SiaStats.info. SiaStats is a non-profit project providing statistics, tools, a blockchain explorer and a network monitor for the Sia network that has been operative since September 2017. It is independent of the Sia Foundation, albeit it works as the de facto network stats and blockchain explorer for the community.

The API is completed and available at: https://siastats.info/navigator-api/web3index/revenue Data on the API is updated 4 times per day (more frequent updates can be included without effort if requested). In compliance with the requirements, the code generating the API is open source and part of the blockchain explorer repository `Navigator` (https://github.com/hakkane84/navigator-sia). The PR including this functionality is: https://github.com/hakkane84/navigator-sia/pull/35

## Can you describe the protocol's tokenomics? Who are the demand and supply side participants in the protocol?

Sia is a traditional Proof of Work blockchain where new coins (SC) are accrued and get in circulation by block-solving, akin to Bitcoin. About 1 day of mining proceedings were pre-mined by the development team in June 2015. Block rewards had a linear decay from 300K coins/block to the current amount of 30k, that represents the bottom of mining inflation. In February 2021, the Sia Foundation was stablished as the steward of the protocol, stablishing an initial subsidy of 1.58 billion, plus additional monthly subsidies that represent 1.58 billion per year. As of October 27th 2021, 46.54 billion coins have been mined by the community and the Foundation subsidies represent 2.63 billion additional coins (5.3% of the total supply), a total of 49.17 billion. The overall inflation will be 9.87% for 2021, 6.34% for 2022 and will progressively decay on the following years. Full figures of coin supply, inflation and their future projections can be found on: https://siastats.info/macroeconomics.

The storage marketplace (i.e. users paying for storage services) is independent from the tokenomics model and is disconnected from the block rewards and inflation. In the Sia protocol, the demand is represented by “renters”, that wish to store data on the cloud, while offer is represented by “hosts”, storage providers of the network. Renters and hosts sign `File contracts` on the blockchain, where: a) the renter deposits an amount of SC as “allowance”, serving as payment, and b) the host deposits another amount as “collateral”, serving as a guarantee for the data availability. Both amounts get locked inside the contract. During the lifetime of the contract, host and renter keep the micropayments for each data operation off-chain, in a payment channel. At the end of the contract the host needs to provide a cryptographic proof that demonstrates it is still preserving the data integrity. This proof closes the payment channel on the blockchain and depending on its outcome, the host will receive its collateral back plus the payment from the renter or both amounts will be burnt.

~(Translating the concept of “network revenue” to the Sia blockchain, possibly the best and most accurate methodology is considering the whole value locked on these file contracts. Contracts represent the real demand of services on the Sia blockchain, as opposed to the total coin supply figures. The amount locked on active contracts gets removed from the circulating supply, hypothetically limiting the formation of new contracts the closer the locked percentage approaches 100%.~

~EDIT:  After the discussion bellow, Revenue in Sia gets re-defined as the amounts paid exclusively by the renter and deposited on the file contract (the `allowance`). The API and the PR on the Navigator blockchain have been updated accordingly.~

**EDIT 2**: After review of the methodology by the Sia Foundation core developer Nate Maninger, Revenue in Sia gets re-defined as the fees paid by the renter to hosts and network minus any fraction of the contract allowance returned to the renter at the end of the contract

**Network revenue gets defined as the as the collection of fees paid by the renters in the file contracts**. 

- Fees paid to the hosts for the storage service. Calculated as the allowance deposited by the renter, minus the unspent part that gets returned to the renter at the end of the contract. Transformed to USD at the exchange rate of the day the contract finalizes
- Fees paid to miners for including the contract in the blockchain and fees paid to SiaFund owners (3.9% of the value of the contract). Transformed to USD at the exchange rate of the day the contract was formed

The API linked above already applies this methodology.

## How much demand-side revenue (non-inflationary or subsidized earnings) did the protocol generate over the past 30 days?

$22591 as of December 20th, 2021 (corrected)

## How much demand-side revenue (non-inflationary or subsidized earnings) did the protocol generate over the past 90 days?

$54077 as of December 20th, 2021 (corrected)

## Why do you believe this protocol is a good fit for The Web3 Index?

Sia is among the most veteran storage projects on the blockchain space and is widely recognized as one of the most decentralized projects by design. The advent of Skynet has granted the protocol new web3 use cases including file sharing, CDN capabilities, decentralized identity (MySky), web3 applications frontend (see [Homescreen](https://homescreen.hns.siasky.net/#/)) or virtual desktop environments (see SkyOS).

As of today, the Sia protocol stores 1.8 PB of data and the network is comprised of 640 hosts in 53 countries, offering a total of 5 PB of capacity. In terms of Skynet adoption, public portals are storing 17.9 million files representing 261TB of data (before considering redundant copies). Propelled by the Skynet development and adoption, the Sia network has experienced during the last 6 months a remarkable growth in terms of used storage (from 0.6 to 1.8 PB, +200%) and contract value / network revenue (from 6.8 to 9.4 million SC locked in contracts, +72%). More key figures can be found at https://siastats.info.

**If included today on the Web 3 Index project, Sia would become the second or third largest protocol in terms of network revenue**.

For all these reasons, I kindly request Sia to be integrated on the Web 3 Index project. I will be pleased answering any question here or at your Discord server, bringing members of the Sia Foundation if needed or modifying the API to accommodate any need.

## Category

"Work Protocol"

## Subcategory

"Storage"

## Website

https://sia.tech/

## Twitter

https://twitter.com/Sia__Foundation

## Coingecko or Coinmarketcap URL

https://www.coingecko.com/en/coins/siacoin

**Comment by hakkane84** [comment]
I wanted to bring in an update regarding the possible methodology that can be used for calculating `network revenue` on Sia, after a very interesting debate I had earlier with other community members on the Sia Discord, that has a broader interest for the Web 3 Index.

Let me first recap that in Sia file contracts are formed by locking Siacoin from two sources: the renter deposits an `allowance` for paying the host, while the host deposits a `collateral` as a guarantee of data integrity and availability. The renter spends the allowance during the lifetime of the contract with micro-payments, kept in a payment channel. At the end of the contract the hosts needs to submit to the blockchain a valid cryptographic proof that it still maintains the data. If this happens, the host gets its collateral returned, plus the part of the allowance that was spent, while the renter gets back any unused part of the allowance. In case no valid proof is provided, the spent allowance and the part of the collateral backing it get burnt, the unused allowance gets returned to the renter and the unused part of the collateral gets returned to the host.

On the API and coin application above I have considered as revenue the whole value of the contracts (both allowance and collateral), based on the idea that the whole amount represents the part of the economy that gets "mobilized" as result of the demand side of the network. Other people argued that an alternative could be considering exclusively the allowance, as this represent how much the renters are actually paying for storage. **In the end, considering which is the best methodology depends on if we consider revenue as `how much economic activity is being driven by the demand side of the network` (current proposal) or `how much users are spending on services (alternative)`**.

I am bringing this debate for honesty and rigor, to help getting the most accurate Sia listing possible and also for helping to define with precision the criteria for other listings.

**Comment by adamsoffer** [comment]
Hey @hakkane84 - appreciate you taking the time to submit this application and for the thoughtfulness around the fee reporting. I'm a big fan of Sia and I look forward to getting it added to the web3 index registry!

> In the end, considering which is the best methodology depends on if we consider revenue as how much economic activity is being driven by the demand side of the network (current proposal) or how much users are spending on services (alternative).

While there's certainly value in reporting demand side "economic activity" the intention is to initially focus on how much users are spending on services. This is the definition we've used for the other integrations. 

Perhaps in the future we can add an additional data point for protocols to showcase "mobilized" or "locked" value since this is certainly an interesting metric and one that a few other protocols listed have in common.

**Comment by hakkane84** [comment]
Hi @adamsoffer. Thanks a lot for the review and the very insightful comments about properly calculating network revenue. Your comments have greatly improved the accuracy and precision of revenue metrics for Sia.

I agree on your proposed methodology. From now on, the API provided by SiaStats considers as revenue exclusively the allowance paid by the Sia renters to form file contracts. This is - how much Sia users are paying for storing data on the Sia network. The collateral locked by the hosts on the same file contracts have been accordingly removed from the revenue calculation.

I have updated the API, the original post opening this issue to reflect the changes, the figures of 30- and 90-day revenue on the same post (updated to ~$45000 for 30-day) and the Sia blockchain explorer PR has been also corrected.

Thanks again, I am looking forward for any further comments and I will be glad to apply any other requested change.

**Comment by hakkane84** [comment]
Hi again @adamsoffer. After a kind reviewing from the Sia Foundation Nate Maninger (@n8maninger, the conversation can be found here: https://github.com/hakkane84/navigator-sia/pull/35), we have decided to change the definition of Revenue for Sia. Previously it was the value of the allowance deposited by the renters on file contracts. However, if at the end of the contract the renter has not spent all these funds on services from the host, the unused allowance gets refunded to the renter. For this reason, we have seen more accurate to substract this returned allowance from the Revenue calculation. In other words, Revenue is now the collection of fees the renters pay during contracts (to hosts, to include the transaction in the blockchain, etc.). 

I have updated the API, the blockchain explorer repository that generates it and edited the opening comment of this Issue (including the 30- and 90-day revenue figures, which are lower now).
